src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/WaWorkflowApiClientToInitiateTask.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.wacaseeventhandler.clients;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.EvaluateDmnRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.EvaluateDmnResponse;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.initiatetask.InitiateTaskDmnRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.initiatetask.InitiateTaskDmnResponse;
+
+@Service
+public class WaWorkflowApiClientToInitiateTask
+    implements WaWorkflowApiClient<InitiateTaskDmnRequest, InitiateTaskDmnResponse> {
+
+    private final RestTemplate restTemplate;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final String workflowApiUrl;
+
+    public WaWorkflowApiClientToInitiateTask(RestTemplate restTemplate,
```

Comment:
Minor one here but would this be better named WorkflowApiClientToInitiateTask no need to specify our team name

Reply:
Hey, love comments about naming :)  Yes, I make this change. I was following the service name which is wa-workflow-API, but yeah, not needed and we don't follow this naming convention in other services either.